### ISSUE TITLE
Broken colorization

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2403]
## Environment data

-   Language Server version: v2022.2.3
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10

## Expected behaviour

Correct colorization when the syntax is valid.

## Actual behaviour

Sometimes code loses colorization and displays white (`append` method in this example).

![image](https://user-images.githubusercontent.com/12860284/154707637-ac837305-b0d6-491a-aa70-476a9b4099f8.png)

Better visible with the GitHub Dark Default color theme:

![image](https://user-images.githubusercontent.com/12860284/154707758-083bad62-20fc-4ccc-aeef-e1418edc6759.png)

## Code Snippet / Additional information

This is somewhat hard to replicate but the surest way to do so I found is pasting code block with incorrect syntax which causes some code to be not colorized (I can live with that), then correcting the syntax error and still having valid code not colorized. Sometimes it's not enough and simply adding/removing blank lines triggers this.

Try pasting the below code snippet in your editor and then remove the slash at the end:
```python
def foo(num: str, flag: bool) -> None:
    stack = list[str]()
    for char in num:
        while flag and stack:
            stack.pop()
        stack.append(char)
/
```
And here's a screen recording that shows the issue:

https://user-images.githubusercontent.com/12860284/154706416-97d3ccb5-7e1c-4222-bc85-c68f2d6b9c6f.mp4

Also noticed that switching the color theme fixes colorization 🤔 I would expect file save to fix it. `Developer: Reload Window` is not enough as well.

**Comment by debonte** [comment]
@Molkree thanks for the repro steps and video. I was able to repro this several times.

Related to #2381?

**Comment by debonte** [comment]
@Molkree I'm OK with keeping this issue open until we have some confidence that they are the same issue. We haven't been able to reproduce #2381.

**Comment by debonte** [comment]
@erictraut this is the issue we talked about last night -- seems to be related to the incomplete type cache. Sometimes we know the type of stack when doing semantic coloring, and sometimes we don't.

**Comment by erictraut** [comment]
Thanks for reporting. This was a bug in the type evaluator. If you're interested in the fix, here's the commit: https://github.com/microsoft/pyright/commit/53a6e184a2e4e168995f0ff4ecb722a9ba4178da.

This will be fixed in the next release.

**Comment by Molkree** [comment]
I believe this was fixed in the [v2022.3.2](https://github.com/microsoft/pylance-release/releases/tag/2022.3.2) release.
